src/tools/merssh/qmakecommand.cpp

```diff
@@ -65,15 +65,44 @@ int QMakeCommand::execute()
         return 1;
     }
 
+    QString evalCommand;
+    QString filterCommand; 
+    const int evalIndex = arguments().indexOf(QLatin1String("-env"));
+    if (evalIndex != -1) {
+        //Creating special pro file
+        const QString ppath = QString("/home/%1/query_var").arg(sshParameters().userName);
+        const int varIndex = evalIndex + 1;
+
+        auto args = arguments();
+        evalCommand = QString("echo \'%1\' > %2;").
+                arg((args.size() > varIndex) ?
+                        //Evaluate var in second parameter
+                        QLatin1String("log($$escape_expand($$eval($$VAR)\\n))") :
+                        //Evaluate all vars
+                        QLatin1String("for(var, $$list($$enumerate_vars())) log($$escape_expand($$var=$$eval($$var)\\n\\r))")).
+                arg(ppath);
+
+        //Tansforming args
+        args.replace(evalIndex, ppath);
+
+        if (args.size() > varIndex)
+            args.replace(varIndex, "VAR=" + args.at(varIndex));
```

Comment:
Wouldn't it be possible to substitute it at c++ level while initializing the temporary pro file?